guides/maintenance.md

```diff
@@ -0,0 +1,45 @@
+## Maintaining the Holochain Scaffolding
+
+This is a guide for maintainers of the Holochain Scaffolding. It may also be useful to contributors wanting to understand how the branching and releases are being done.
+
+### Branching strategy
+
+New development and bug fixes should target the `develop` branch. From there, changes are back-ported to maintanance branches which are named after the version of Holochain they work with. So maintance for the versions of Scaffolding that work with Holochain 0.1.x are done on `develop-0.1` and those that work with Holochain 0.2.x are on `develop-0.2` and so on.
+
+There need to be some exceptions to this workflow to allow changes on maintenance branches that don't make sense to make on `develop`:
+
+- Bug fixes that are specific to a particular version of Holochain. This should be rare, but if Holochain were to introduce an API change between 0.1.x and 0.1.x+1 but NOT make that change for Holochain 0.2.x then the PR for Scaffolding would target `develop-0.1` directly.
+- Version bumps for `flake.nix`, `flake.lock` and `Cargo.lock` files should target the maintenance branches directly.
+- Version bumps for dependencies of Scaffolding in the `Cargo.toml`. For dependencies that still on the same version on `develop` then it would be valid to target `develop` and back-port but you'll end up with a tricky lock file merge anyway.
```

Comment:
```suggestion
- Version bumps for dependencies of Scaffolding in the `Cargo.toml`. For dependencies that are still on the same version as on `develop` it would be valid to target `develop` and back-port but you'll end up with a tricky lock file merge anyway.
```